v2/tools/generator/internal/astbuilder/builder.go

```diff
@@ -255,6 +255,38 @@ func ReturnIfExpr(cond dst.Expr, returns ...dst.Expr) *dst.IfStmt {
 	}
 }
 
+// CheckIfNotNilAndSingleStatement checks if the ident is non-nil, and if it is executes the provided statement.
+//
+//	if <ident> != nil {
+//		<stmt>
+//	}
+func CheckIfNotNilAndSingleStatement(stmt dst.Stmt, ident string) dst.Stmt {
+	return &dst.IfStmt{
+		Cond: NotNil(dst.NewIdent(ident)),
+		Body: &dst.BlockStmt{
+			List: []dst.Stmt{
+				stmt,
+			},
+		},
+	}
+}
```

Comment:
Suggest putting the identifier first as that reflects the order both the generated code and the identifier.

Actually, you could easily make `stmt` varidic and make it more flexiible:

```suggestion
func CheckIfNotNilAndSingleStatement(ident string, stmts ...dst.Stmt) dst.Stmt {
	return &dst.IfStmt{
		Cond: NotNil(dst.NewIdent(ident)),
		Body: BlockStatement(stmts),
	}
}
```

Reply:
And ditto below.